### PR TITLE
test(workbenches): add notebook controller deployment health checks

### PR DESCRIPTION
**Pull request by jstourac**
Verify that odh-notebook-controller and notebook-controller (KF) deployments have the expected replica count, container count, and manager container resource requests/limits.

Migrated from ods-ci 0500__ide_deployment_cr_checks.robot as pure CLI/API tests using openshift-python-wrapper.

## Related Issues

[RHOAIENG-68943](https://redhat.atlassian.net/browse/RHOAIENG-68943)

## Please review and indicate how it has been tested

```
uv run pytest tests/workbenches/notebooks_server/controller/test_deployment_health.py
```

- [x] Locally
- [ ] Jenkins

## Additional Requirements

- [ ] If this PR introduces a new test image, did you create a PR to mirror it in disconnected environment?
- [ ] If this PR introduces new marker(s)/adds a new component, was relevant ticket created to update relevant Jenkins job?

[RHOAIENG-68943]: https://redhat.atlassian.net/browse/RHOAIENG-68943?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **Tests**
  * Added a new smoke test suite to validate notebook controller deployment health in the Kubernetes applications namespace.
  * Checks that the expected replica count is up and pods are in a running state (excluding terminating pods).
  * Verifies each controller’s pod container count and confirms the manager container CPU/memory requests and limits match expected values.

### CODE REVIEW

**Comment by jiridanek** [comment]
@coderabbitai full review

**Review by jiridanek — DISMISSED** [review]

File: tests/workbenches/notebooks_server/controller/test_deployment_health.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+        pods = list(
+            Pod.get(
+                client=admin_client,
+                namespace=namespace,
+                label_selector=controller_config["label_selector"],
+            )
+        )
+
+        assert len(pods) == EXPECTED_REPLICAS, (
```
Comment (harshad16) [inline tests/workbenches/notebooks_server/controller/test_deployment_health.py:61]: I believe there is an edge case, 
there is terminating pod, along with running pod, this could be missed.
wdyt? also i believe we can fix it in follow up PR.

**Review by harshad16 — COMMENTED** [review]
/lgtm

**Review by harshad16 — DISMISSED** [review]

File: tests/workbenches/notebooks_server/controller/test_deployment_health.py (hunk 1 above)
Comment (jstourac) [reply tests/workbenches/notebooks_server/controller/test_deployment_health.py:61]: great catch! updated, thank you!

**Review by harshad16 — APPROVED** [review]

**Review by jiridanek — APPROVED** [review]
